app/helpers/jobseekers/job_applications_helper.rb

```diff
@@ -0,0 +1,33 @@
+module Jobseekers::JobApplicationsHelper
+  def application_link_path
+    if jobseeker_application_status.nil?
+      new_jobseekers_job_job_application_path(@vacancy.id)
+    elsif jobseeker_application_status == "submitted"
+      jobseekers_job_application_path(job_application.id)
+    else
+      jobseekers_job_application_review_path(job_application.id)
+    end
+  end
+
+  def application_link_text
+    if jobseeker_application_status.nil?
+      "apply"
+    elsif jobseeker_application_status == "submitted"
+      "submitted"
+    else
+      "draft"
+    end
+  end
+
+  def jobseeker_has_not_signed_in_or_applied?
```

Comment:
This decides whether we show the 'save job' button, right?

I _think_ we only need to check if they have applied, not whether they're signed in.

Reply:
Also, might it be more readable to write this method as a positive question: `unless jobseeker_has_applied?` vs. `if jobseeker_has_not_applied`?